internal/service/management_api.go

```diff
@@ -0,0 +1,60 @@
+package service
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	metrics "github.com/slok/go-http-metrics/metrics/prometheus"
+	"github.com/slok/go-http-metrics/middleware"
+	"github.com/slok/go-http-metrics/middleware/std"
+	"github.com/topfreegames/maestro/internal/config"
+	"github.com/topfreegames/maestro/internal/core/monitoring"
+	"github.com/topfreegames/maestro/internal/handlers"
+	api "github.com/topfreegames/maestro/protogen/api/v1"
+	"go.uber.org/zap"
+)
+
+func ProvideManagementMux(ctx context.Context, pingHandler *handlers.PingHandler) *runtime.ServeMux {
+	mux := runtime.NewServeMux()
+	api.RegisterPingHandlerServer(ctx, mux, pingHandler)
+
+	return mux
+}
+
+// RunManagementServer starts HTTP server in other goroutine, and returns a
+// shutdown function. It serves management API endpoints/handlers.
+func RunManagementServer(ctx context.Context, configs config.Config, mux *runtime.ServeMux) func() error {
+	if !configs.GetBool("management_api.enabled") {
```

Comment:
Not sure if we want this kind of option since the `management_api` is a command. If we run the `cmd/management_api` with this config as `false`, it won't do anything.